### ISSUE TITLE
XML Deserialization Issues

### ISSUE THREAD

**Issue by ChrisH84** [lordmilko/PrtgAPI#1]
Hi,

Thank you very much for your contribution, I think this is a great project.
I've encountered several XMLDeserialization Errors while pulling data from PRTG:

For example if all Sensors are queried, there is a deserialization error while trying to parse the Column "Access" when it is empty.

Another example is that the Input String was not in correct format if you try to pull all the devices from PRTG.

Will you be fixing these Issues or should I fix them by myself ?

I just wanted to inform you about this minor bug.

Thank you very much in advance and best regards,
Chris

**Comment by lordmilko** [comment]
Hi Chris,

Thanks for your bug report! I have identified an issue wherein 0 was a valid value for the Priority field, however I have been unable to replicate your errors with the Access column, likely as none of my sensors have a missing "Access" property (under normal circumstances you would expect this is not considered valid behaviour). If you set the property Access in Objects\SensorOrDeviceOrGroupOrProbe.cs to Access? and then LINQ to find the sensor you have with a missing Access value, I would be interested to see what PRTG has to say about its setting if you go to that sensor in the web interface.

In regards to the GetDevices call, are you able to tell me what the InnerException was? This may tell us the field that could not be deserialized. Failing that, if you set a breakpoint on line 134 of PrtgClient.cs in GetInvalidXml you might be able to see the failing field as well. (In theory this method is meant to deconstruct the XML Deserializer's InvalidOperationException and tell you what is actually going on, however I have discovered a separate bug with this method under certain circumstances I am currently fixing as well; setting the breakpoint will at the very least allow you to check the value of the xml variable in the event the method crashes.)
